client/src/components/MapGeocoderWrapper.vue

```diff
@@ -0,0 +1,106 @@
+<template>
+  <div v-show='visible' class='geocoder-content-expanded'>
+    <div class='geocoder' id='geocoder'></div>
+    <div class='search-button' @click='this.visible = !this.visible'>
+      <img src='@/assets/icons/search.svg' />
+    </div>
+  </div>
+  <div v-show='!visible' class='geocoder-content-collapsed'>
+    <div class='search-button' @click='this.visible = !this.visible'>
+      <img src='@/assets/icons/search.svg' />
+    </div>
+  </div>
+</template>
+
+<script lang='ts'>
+import { defineComponent, inject } from 'vue';
+import { State } from '../model/state';
+import { stateKey } from '../injection_keys';
+import MapboxGeocoder from '@mapbox/mapbox-gl-geocoder';
+import '@mapbox/mapbox-gl-geocoder/dist/mapbox-gl-geocoder.css';
+import mapboxgl from 'mapbox-gl';
+
+/**
+ * Wrapper for Mapbox Geocoder.
+ *
+ * See details at https://github.com/mapbox/mapbox-gl-geocoder.
+ */
+export default defineComponent({
+  name: 'MapGeocoderWrapper',
+  setup() {
+    const state: State = inject(stateKey, State.default());
+
+    return {
+      state,
+    };
+  },
+  data() {
+    return {
+      geocoder: null as unknown as MapboxGeocoder,
+      visible: false,
+    };
+  },
+  watch: {
+    'state.map': function(newMap: mapboxgl.Map) {
+      // Create geocoder when map is updated and geocoder is null. This is only triggered once per
+      // app load since the map is only created once.
+      if (newMap != null && this.geocoder == null) {
+        this.geocoder = new MapboxGeocoder({
+          accessToken: process.env.VUE_APP_MAP_BOX_API_TOKEN,
+          mapboxgl: mapboxgl as unknown as mapboxgl.Map,
```

Comment:
Same as above regarding `as unknown` here.

Reply:
This is necessary if we want a marker. This actually seems to be a bug with the geocoder code - it requires the type  mapboxgl.Map but expects just mapboxgl type (see example: https://docs.mapbox.com/mapbox-gl-js/example/mapbox-gl-geocoder-outside-the-map/). 

I'll add a comment to clarify. Also, this is only required if we want the marker placed on the map when it zooms to a location. Otherwise, this can be left undefined.

Reply:
I don't see how that example requires this casting. What is `mapboxgl` itself? It seems like a library, while the `MapboxGeocoder` says it wants "A mapbox-gl instance to use when creating Markers". Can we instantiate something to pass?

I'm also fine skipping markers for now unless you use them.

Reply:
The example is just js, not typescript so the casting isn't necessary. I think there is a big with the mapbox geocoder types which require what you mention "an instance". I tried passing the map instance and got this issue:

https://stackoverflow.com/questions/57467842/getting-error-this-mapboxgl-marker-is-not-a-constructor-when-using-geocoder

I'll just remove it for now since it isn't on mocks and this is confusing.